File: SERVICES/Services1/app/src/main/java/com/javadude/services1/StartedService.java
package com.javadude.services1;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.support.annotation.Nullable;
import android.util.Log;

public class StartedService extends Service {
	@Override
	public void onCreate() {
		super.onCreate();
		Log.d("StartedService", "onCreate");
	}

	private Thread counterThread = new Thread() {
		@Override public void run() {
			for(int i = 1; !isInterrupted() && i <=100; i++) {
				Log.d("StartedService", "count = " + i);
				try {
					Thread.sleep(500);
				} catch (InterruptedException e) {
					interrupt();
				}
			}
			stopSelf();
		}
	};

	@Override
	public int onStartCommand(Intent intent, int flags, int startId) {
		Log.d("StartedService", "onStartService");
		counterThread.start();
		return super.onStartCommand(intent, flags, startId);
	}

	@Override
	public void onDestroy() {
		super.onDestroy();
		counterThread.interrupt();
		Log.d("StartedService", "onDestroy");
	}

	@Nullable @Override public IBinder onBind(Intent intent) {
		return null;
	}
}

File: Module - Services/Services1/app/src/main/java/com/javadude/services1/StartedService.kt
package com.javadude.services1

import android.app.Service
import android.content.Intent
import android.os.Binder
import android.util.Log

// A started service (basically "start and forget")
class StartedService : Service() {
    private val counterThread = object : Thread() {
        override fun run() {
            var i = 0
            while(!isInterrupted && i <= 100) {
                Log.d("StartedService", "count = $i")
                try {
                    sleep(250)
                } catch (e : InterruptedException) {
                    interrupt()
                }

                i++
            }
            stopSelf()
        }
    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        counterThread.start() // start some processing
        return START_REDELIVER_INTENT // try to re-run service later if killed
    }

    override fun onDestroy() {
        counterThread.interrupt()
        super.onDestroy()
    }
    override fun onBind(intent: Intent?): Binder? = null
}